src/subscribers/redis-subscriber.ts

```diff
@@ -36,7 +36,9 @@ export class RedisSubscriber implements Subscriber {
                         Log.info("Event: " + message.event);
                     }
 
-                    callback(channel, message);
+                    const internalChannel = this.options.redisChannelPrefix ? channel.replace(new RegExp(`^${this.options.redisChannelPrefix}`), '') : channel;
```

Comment:
I like that this is simple by not touching a lot of code, but is using regexp performant enough for potentially higher traffic sites?

Reply:
I'm sorry. I have not yet consider the performance impact. In fact, I'm wondering how can I do a test about this. If you got some idea,  let me know. I would like to do a test.

Reply:
I tried doing a simple test with a loop for the line in question:

```javascript
var start = Date.now();

var redisChannelPrefix = 'laravel_database_';
var channel = 'laravel_database_private-user.1234';
var i;

for (i = 0; i < 1000000; i++) {
  var result = redisChannelPrefix ? channel.replace(new RegExp(`^${redisChannelPrefix}`), '') : channel;
}

console.log(Date.now() - start);
```

It shows that it takes around 300 milliseconds (0.3 seconds) to run the RegExp line 1.000.000 times.

In my book that is fine.

Reply:
Cool! I like this, but it's obviously up to the maintainer then